src/services/expenses.js

```diff
@@ -0,0 +1,67 @@
+'use strict';
+
+let expenses = [];
+
+function setInitialExpenses() {
+  expenses = [];
+}
+
+function getAll(userId, category, from, to) {
+  return expenses.filter(expense => {
+    const isUserIdMatch = userId
+      ? expense.userId === +userId
+      : true;
+
+    const isCategoriesMatch = category
+      ? expense.category === category
+      : true;
+
+    const isFromMatch = from
+      ? expense.spentAt >= from
+      : true;
+
+    const isToMatch = to
+      ? expense.spentAt <= to
+      : true;
+
+    return isUserIdMatch && isCategoriesMatch && isFromMatch && isToMatch;
+  });
+};
+
+function addExpense(expenseData) {
+  const newExpense = {
+    id: expenses.length + 1,
```

Comment:
I think you have to use a separate counter for Id generating, because now if you add 5 expenses to collection, then you remove first two expenses and after removing if you add new expense it will be with Id 4 (not 6). There will two expenses with Id 4 in the collection.